hotshot-state-prover/src/service.rs

```diff
@@ -315,7 +368,37 @@ pub async fn submit_state_and_proof(
     // prepare the input the contract call and the tx itself
     let proof: ParsedPlonkProof = proof.into();
     let new_state: ParsedLightClientState = public_input.into();
-    let tx = contract.new_finalized_state(new_state.into(), proof.into());
+
+    let mut tx = contract.new_finalized_state(new_state.into(), proof.into());
+
+    // TODO: (alex) centralize this to a dedicated ethtxmanager with gas control
+    // frugal gas price: set to SafeGasPrice based on live price oracle
+    // safe to be included with low priority position in a block.
+    // ignore this if etherscan fail to return
+    if let Ok(res) =
+        reqwest::get("https://api.etherscan.io/api?module=gastracker&action=gasoracle").await
+    {
+        // parse etherscan response,
+        // if etherscan API changes, we log the warning and fallback to default (by falling through)
+        if let Ok(res) = res.json::<EtherscanPriceOracleResponse>().await {
+            tracing::info!("Etherscan gas oracle: {:?}", res.result);
+            match TryInto::<GasInfo>::try_into(res.result) {
+                Err(e) => {
+                    tracing::warn!("{}", e)
+                }
+                Ok(gas_info) => {
+                    if let TypedTransaction::Eip1559(inner) = &mut tx.tx {
```

Comment:
Are we sure this is the right transaction variant returned by the `new_finalized_state` binding? I feel like if the transaction variant doesn't match we should at least log something

Reply:
Yes it's our tx type. All tx after  London upgrade (since [Aug 2021](https://ethereum.org/en/history/#london)) is this type.